From 3db311a02d90286ba9d23c5d0ccc4bd26a46f84f Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:53:35 -0400
Subject: [fs] ext4: convert write_begin methods to stable_page_writes semantics

Message-id: <1414058125-4183-62-git-send-email-lczerner@redhat.com>
Patchwork-id: 98152
O-Subject: [RHEL 7.1 PATCH v2 061/171] ext4: convert write_begin methods to stable_page_writes semantics
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Dmitry Monakhov <dmonakhov@openvz.org>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit 7afe5aa59ed3da7b6161617e7f157c7c680dc41e

 Use wait_for_stable_page() instead of wait_on_page_writeback()

 Signed-off-by: Dmitry Monakhov <dmonakhov@openvz.org>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Reviewed-by: Jan Kara <jack@suse.cz>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/inode.c b/fs/ext4/inode.c
index f91e0b2..5b1326c 100644
--- a/fs/ext4/inode.c
+++ b/fs/ext4/inode.c
@@ -969,7 +969,8 @@ retry_journal:
   ext4_journal_stop(handle);
   goto retry_grab;
  }
- wait_on_page_writeback(page);
+ /* In case writeback began while the page was unlocked */
+ wait_for_stable_page(page);
 
  if (ext4_should_dioread_nolock(inode))
   ret = __block_write_begin(page, pos, len, ext4_get_block_write);
@@ -2679,7 +2680,7 @@ retry_journal:
   goto retry_grab;
  }
  /* In case writeback began while the page was unlocked */
- wait_on_page_writeback(page);
+ wait_for_stable_page(page);
 
  ret = __block_write_begin(page, pos, len, ext4_da_get_block_prep);
  if (ret < 0) {
-- 
1.7.1